source/extensions/filters/common/ext_authz/check_request_utils.cc

```diff
@@ -114,20 +114,29 @@ void CheckRequestUtils::setHttpRequest(
         return Envoy::Http::HeaderMap::Iterate::Continue;
       },
       mutable_headers);
+
+  // Set request body.
+  const Buffer::Instance* buffer = sdfc->decodingBuffer();
+  if (max_request_bytes > 0 && buffer != nullptr) {
+    const uint64_t len = std::min(buffer->length(), max_request_bytes);
+    std::unique_ptr<char[]> data(new char[len]);
```

Comment:
I think I could use an `std::string` and then just move it. Not sure.

Reply:
yeah I think you can do that.